Sources/ITwinMobile/ITMApplication.swift

```diff
@@ -269,12 +296,43 @@ open class ITMApplication: NSObject, WKUIDelegate, WKNavigationDelegate {
         usingRemoteServer = false
         return "imodeljs://app/index.html"
     }
+    
+    /// Converts the given hash parameters into a URL hash string, encoding values so that they are valid for use in a URL.
+    /// - Parameter hashParams: The hash parameters to convert.
+    /// - Returns: The hash parameters converted to a URL hash string.
+    public func stringifyHashParams(_ hashParams: HashParams) -> String {
+        var result = ""
```

Comment:
Here's a more compact implementation if you'd like to use it:

```swift
public func stringifyHashParams(_ hashParams: HashParams) -> String {
    let allowedCharacters = CharacterSet(charactersIn: "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_.")
    let encoded = hashParams.map { "\($0.name)=\($0.value.addingPercentEncoding(withAllowedCharacters: allowedCharacters)!)" }
    return "#" + encoded.joined(separator: "&")
}
```

Reply:
Actually, maybe it would be better to make it an extension function:

```swift
extension HashParams {
    func stringify() -> String {
        let allowedCharacters = CharacterSet(charactersIn: "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_.")
        let encoded = self.map { "\($0.name)=\($0.value.addingPercentEncoding(withAllowedCharacters: allowedCharacters)!)" }
        return "#" + encoded.joined(separator: "&")
    }
}
```

Reply:
I agree with both suggestions, although I will use toString() instead of stringify().